control/grpc.py

```diff
@@ -0,0 +1,446 @@
+#
+#  Copyright (c) 2021 International Business Machines
+#  All rights reserved.
+#
+#  SPDX-License-Identifier: LGPL-3.0-or-later
+#
+#  Authors: anita.shekar@ibm.com, sandy.kaur@ibm.com
+#
+
+import socket
+import grpc
+import json
+import logging
+from google.protobuf import json_format
+from .generated import gateway_pb2 as pb2
+from .generated import gateway_pb2_grpc as pb2_grpc
+
+
+class GatewayService(pb2_grpc.GatewayServicer):
+    """Implements gateway service interface.
+
+    Handles configuration of the SPDK NVMEoF target according to client requests.
+
+    Instance attributes:
+        config: Basic gateway parameters
+        logger: Logger instance to track server events
+        gateway_name: Gateway identifier
+        gateway_state: Methods for target state persistence
+        spdk_rpc: Module methods for SPDK
+        spdk_rpc_client: Client of SPDK RPC server
+    """
+
+    def __init__(self, config, gateway_state, spdk_rpc, spdk_rpc_client):
+
```

Comment:
Is this missing `logging.basicConfig(level=logging.DEBUG)` like `server.py` used to?

Reply:
`logging.basicConfig(level=logging.DEBUG)` configures the root logger that `server.py`, `state.py`, and `grpc.py`'s loggers inherit from so setting it once in `server.py` was sufficient for setting the log level. I changed this to set up the root logger in `main.py` to be more straightforward and because it makes more sense to set up more advanced logging features there.